src/main/java/frc/robot/subsystems/Intake.java

```diff
@@ -1,35 +1,66 @@
 package frc.robot.subsystems;
 
+import com.chopshop166.chopshoplib.outputs.IDSolenoid;
 import com.chopshop166.chopshoplib.outputs.SendableSpeedController;
 
+import edu.wpi.first.wpilibj.DoubleSolenoid.Value;
 import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
 import edu.wpi.first.wpilibj2.command.StartEndCommand;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 import frc.robot.maps.RobotMap;
 
+// The Intake takes in the balls from the ground and it is raised and retracted by pneumatic pistons 
+// There is only one speed for the intake motor (Member: Motor) 
+// The intake interacts with the Indexer It uses motors, wheels, and pistons connected by belts and bands(The controller) 
+// The intake needs to make sure the balls pass through it and don't get jammed(Member: Motor)
+
 public class Intake extends SubsystemBase {
-    private SendableSpeedController rollerMotor;
+    private final SendableSpeedController rollerMotor;
+    private final IDSolenoid deployPiston;
+    private final IDSolenoid rollerPlacement;
 
     private static final double rollerMotorSpeed = 0.85;
 
-    public Intake(RobotMap.IntakeMap map) {
+    public Intake(final RobotMap.IntakeMap map) {
         super();
         rollerMotor = map.roller();
+        deployPiston = map.deployPiston();
+        rollerPlacement = map.deployIntake();
     }
 
-    private CommandBase runRoller(double motorSpeed) {
+    // pneumatic pistons to raise the intake up so that balls can get underneath
+
+    public CommandBase runRoller() {
         return new StartEndCommand(() -> {
-            rollerMotor.set(motorSpeed);
+            rollerMotor.set(rollerMotorSpeed);
+            rollerPlacement.set(Value.kForward);
         }, () -> {
             rollerMotor.stopMotor();
+            rollerPlacement.set(Value.kReverse);
         }, this);
     }
 
     public CommandBase intake() {
-        return runRoller(rollerMotorSpeed);
+        return runRoller();
```

Comment:
Kind of like how `deployPiston` and `retractPiston` use the same function with different arguments, make `runRoller` take a speed argument and pass different values in from `intake` and `discharge`. Right now both commands run the motors in the same direction.